source/main.cpp

```diff
@@ -365,160 +357,34 @@ WMMSettingsView::~WMMSettingsView()
 	delete fAuthorView;
 
 	delete Icon;
-
-	SaveSettings( &Settings);
 }
 
 //---------------------------------------------------
-//	Attached to window - set siblings target to this
+//	Attached to window - set siblings target
 //---------------------------------------------------
 void
 WMMSettingsView::AttachedToWindow()
 {
-	slidRadius->SetTarget( this);
-	slidCircles->SetTarget( this);
-	slidLineWidth->SetTarget( this);
-	slidLineSpace->SetTarget( this);
-	slidPulseRate->SetTarget( this);
-	Window()->SetPulseRate( Settings.pulse);
+	slidRadius->SetTarget(fDemoView);
+	slidCircles->SetTarget(fDemoView);
+	slidLineWidth->SetTarget(fDemoView);
+	slidLineSpace->SetTarget(fDemoView);
+	slidPulseRate->SetTarget(fDemoView);
 }
 
 //---------------------------------------------------
 //	Draw view
 //---------------------------------------------------
 void
-WMMSettingsView::Draw( BRect updateRect)
-{
-	demoRect = fDemoView->Bounds();
-
-	//demo background
-	fDemoView->SetHighColor(255, 255, 255, 255);
-	fDemoView->FillRoundRect(demoRect, 2, 2);
-	fDemoView->SetHighColor(0, 0, 0, 255);
-	fDemoView->StrokeRoundRect(demoRect, 2, 2);
-
-	InitDemo();
-}
-
-//---------------------------------------------------
-//	Draw demo
-//---------------------------------------------------
-void
-WMMSettingsView::DrawDemo()
+WMMSettingsView::Draw(BRect updateRect)
 {
-	float right = demoRect.right, bottom = demoRect.bottom;
-
-	if( demoCircle < Settings.circles)
-	{
-		float oldPen = fDemoView->PenSize();
-		fDemoView->SetPenSize(Settings.lwidth);
-		BRegion region = BRegion(BRect(1, 1, right - 1, bottom - 1));
-		fDemoView->ConstrainClippingRegion(&region);
-
-		fDemoView->SetHighColor(demoColor);
-		fDemoView->StrokeEllipse(BPoint(right / 2, bottom / 2), demoRadius, demoRadius);
-		demoRadius += ( Settings.lwidth / 2) + Settings.lspace;
-		demoColor.alpha += demoAlphaMod;
-		demoCircle++;
-
-		fDemoView->ConstrainClippingRegion(NULL);
-		fDemoView->SetPenSize(oldPen);
-	}
-	else
-	{
-		//demo background
-		fDemoView->SetHighColor(255, 255, 255, 255);
-		fDemoView->FillRoundRect(BRect(1, 1, right - 1, bottom - 1), 2, 2);
-		//SetHighColor( 0, 0, 0, 255);
-		//StrokeRoundRect( demoRect, 2, 2);
-
-		InitDemo();
-	}
+	fDemoView->Invalidate();
```

Comment:
I think you already have the required variable (demoCircle) and clipping (ConstrainClippingRegion).

What I would do is:
- Draw() will always draw the background rectangle, then the frame, then the circles up to demoCircle (for(i = 0; i < demoCircle; i++) ...). This way, the complete view is refreshed.
- Pulse() will change the value of demoCircle (increment, then reset to 0) each time it is called, then Invalidate() to make sure the view is redrawn at least once with the new circle.

Also, you can probably rename demoRadius, demoColor, demoCircle, etc to fRadius, fColor, fCircle, since now they are part of the DemoView and it is clear they are used for the demo only.